core/src/test/java/org/apache/iceberg/view/TestViewMetadata.java

```diff
@@ -0,0 +1,280 @@
+        .hasMessage(
+            "Cannot build ViewMetadata, some of required attributes are not set [location, currentSchemaId, currentVersionId]");
+
+    assertThatThrownBy(
+            () -> ImmutableViewMetadata.builder().formatVersion(1).location("location").build())
+        .isInstanceOf(IllegalStateException.class)
+        .hasMessage(
+            "Cannot build ViewMetadata, some of required attributes are not set [currentSchemaId, currentVersionId]");
+
+    assertThatThrownBy(
+            () ->
+                ImmutableViewMetadata.builder()
+                    .formatVersion(1)
+                    .location("location")
+                    .currentSchemaId(1)
+                    .build())
+        .isInstanceOf(IllegalStateException.class)
+        .hasMessage(
+            "Cannot build ViewMetadata, some of required attributes are not set [currentVersionId]");
+
+    assertThatThrownBy(
+            () ->
+                ImmutableViewMetadata.builder()
+                    .formatVersion(1)
+                    .location("location")
+                    .currentSchemaId(1)
+                    .currentVersionId(1)
+                    .build())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid view versions: empty");
+  }
+
+  @Test
+  public void unsupportedFormatVersion() {
+    assertThatThrownBy(
+            () ->
+                ImmutableViewMetadata.builder()
+                    .formatVersion(23)
+                    .location("location")
+                    .currentSchemaId(1)
+                    .currentVersionId(1)
+                    .build())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Unsupported format version: 23");
+  }
+
+  @Test
+  public void emptyViewVersion() {
+    assertThatThrownBy(
+            () ->
+                ImmutableViewMetadata.builder()
+                    .formatVersion(1)
+                    .location("location")
+                    .currentSchemaId(1)
+                    .currentVersionId(1)
+                    .build())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid view versions: empty");
+  }
+
+  @Test
+  public void emptySchemas() {
+    assertThatThrownBy(
+            () ->
+                ImmutableViewMetadata.builder()
+                    .formatVersion(1)
+                    .location("location")
+                    .currentSchemaId(1)
+                    .currentVersionId(1)
+                    .addVersions(
+                        ImmutableViewVersion.builder()
+                            .schemaId(1)
+                            .versionId(1)
+                            .timestampMillis(23L)
+                            .putSummary("operation", "op")
+                            .build())
+                    .addHistory(
+                        ImmutableViewHistoryEntry.builder()
+                            .timestampMillis(23L)
```

Comment:
Minor: Tests should use valid values to avoid problems later when values are checked.

Reply:
I've updated the values used for format version, current schema/version id. Does it matter if we use `System.currentTimeMillis()` in these tests here? It just adds less predictable values (and also we don't care about the current timestamp anywhere in the tests)

Reply:
As long as we're not doing validation, it should be fine.